code/modules/mob/living/carbon/xenomorph/egg.dm

```diff
@@ -0,0 +1,202 @@
+/obj/effect/alien/egg
+	name = "theoritical egg"
+	density = FALSE
+	flags_atom = CRITICAL_ATOM
+	max_integrity = 80
+	///What maturity stage are we in
+	var/maturity_stage = 0
+	///Time between two maturity stages
+	var/maturity_time = 0
+	///Number of the last maturity stage before bursting
+	var/stage_ready_to_burst = 0
+	///Which hive it belongs to
+	var/hivenumber = XENO_HIVE_NORMAL
+	///How far will targets trigger the burst
+	var/trigger_size = 0
+
+/obj/effect/alien/egg/Initialize(mapload, hivenumber)
+	. = ..()
+	src.hivenumber = hivenumber
+	advance_maturity(maturity_stage)
+
+/obj/effect/alien/egg/update_icon_state()
+	icon_state = initial(icon_state) + "[maturity_stage]"
+
+///Advance the maturity state by one, or set it to maturity
+/obj/effect/alien/egg/proc/advance_maturity(maturity)
+	maturity_stage = maturity ? maturity : maturity_stage + 1
+	update_icon()
+	if(maturity_stage < stage_ready_to_burst)
+		addtimer(CALLBACK(src, .proc/advance_maturity), maturity_time)
+		return
+	if(maturity_stage != stage_ready_to_burst)
+		return
+	for(var/turf/turf_to_watch AS in filled_turfs(src, trigger_size, "circle", FALSE))
+		RegisterSignal(turf_to_watch, COMSIG_ATOM_ENTERED, .proc/enemy_crossed)
+
+///Bursts the egg.
+/obj/effect/alien/egg/proc/burst(kill)
+	SHOULD_CALL_PARENT(TRUE)
+	if(kill)
+		advance_maturity(stage_ready_to_burst + 2)
+	else
+		advance_maturity(stage_ready_to_burst + 1)
+	for(var/turf/turf_to_watch AS in filled_turfs(src, trigger_size, "circle", FALSE))
+		UnregisterSignal(turf_to_watch, COMSIG_ATOM_ENTERED)
+
+///Signal handler to check if the atom moving nearby is an enemy
+/obj/effect/alien/egg/proc/enemy_crossed(datum/source, atom/movable/entered)
+	SIGNAL_HANDLER
+	if(!iscarbon(entered))
+		return
+	if(!should_proc_burst(entered))
+		return
+	burst()
+
+/obj/effect/alien/egg/proc/should_proc_burst(mob/living/carbon/carbon_mover)
+	if(issamexenohive(carbon_mover))
+		return FALSE
+	if(carbon_mover.stat == DEAD)
+		return FALSE
+	return TRUE
```

Comment:
this should probably use the hugger inside or whatever proc to check if it should attack the enemy, this way its simpler for multiple hugger types.

Reply:
that requires instantiating the hugger and keeping it in content. Which is really meh